### DISCUSSION TITLE
Idiomatic way to find intersection between a circular shape and a polygon in s2geometry?

### DISCUSSION THREAD

**Discussion by sreekanth-cb** [google/s2geometry#231] [Q&A] [answered]
I was trying to represent a circular shape on a 2D map with s2cap, 
but then wasn't getting the right results while checking the intersection with a polygon. 
Any examples/guidance would be highly helpful.

sample code
```
cp := s2.PointFromLatLng(s2.LatLngFromDegrees(c.Vertices[1], c.Vertices[0]))
angle := radiusInMetersToS1Angle(float64(c.Radius))
s2cap := s2.CapFromCenterAngle(cp, angle)

cellIDs := s2Polygon.CellUnionBound()
for _, cellID := range cellIDs {
	if s2cap.IntersectsCell(s2.CellFromCellID(cellID)) {
		return true, nil
	}
}

```

**Comment by jmr** [comment, marked as answer]
Well, it appears you're using the go version, but the principle is the same.

If you just have one polygon and one point, then `S2Point projected = polygon.Project(cap_center);` and test the distance between `projected` and `cap_center`.  If you have multiple polygons and want to test if any are within the distance, then the idiomatic way is probably with an `S2ShapeIndex`, but that's more complicated.

**Comment by sreekanth-cb** [comment]
@jmr, thanks a lot for the quick responses. 
Let me try this out, but I need to port both Project & ProjectToBoundary polygon APIs to the Go version.
and let me get back with my findings.

**Comment by sreekanth-cb** [comment]
It works for my current use case and closing this ticket! 
Thanks a lot @jmr!!I

**Comment by sreekanth-cb** [comment]
@jmr,

Can you please help me with another doubt here,

what would be the best s2geometry native shape representing a geojson shape like "linestring/multilinestring" ? 
loop/[]loop Or 
a polyline?

thanks,

**Comment by jmr** [comment]
`LineString` ~ `S2Polyline`. There's nothing like `MultiLineString`. `vector<S2Polyline>`, I suppose.

An `S2Loop` is a simple polygon.

**Reply by sreekanth-cb** [reply]
Thanks, @jmr!
